### PR TITLE
Implement recipe filtering

### LINKED ISSUE #16: As an authenticated user, I want to be able to filter my recipe search results
**Acceptance Criteria:**
* On the search screen, there are dropdown filters under the text field that allow me to filter the search result that are based on the meal cost and preparation time.
* Meal cost filter option should have a list of checkboxes that has an inclusive range of meal cost, such as:
          * $0 - $20
          * $20 - $50
          * $50 - $100
          * $100+
* Preparation time filter option should have a list of checkboxes that has an inclusive time range  for the meal preparation:
          * 0 - 25 minutes
          * 25 - 45 minutes
          * 45 - 60 minutes
          * 60 minutes +
* When I select any of the filters, the search results should be updated.

### PR DESCRIPTION
**Pull request by adurso3**
Closes #16 

Note: recipe filtering is only implemented for search results and not for the default suggested meals, since the search endpoint requires a query.

### CODE REVIEW

File: app/src/main/java/com/mealsmadeeasy/ui/home/discover/DiscoverFragment.kt [hunk 1]
```diff
@@ -18,7 +18,7 @@ class DiscoverFragment : BaseFragment() {
     @Inject lateinit var mealStore: MealStore
 
     companion object {
-        private const val TAG = "DiscoverFragment"
+        const val TAG = "DiscoverFragment"
```
Comment (andrewbailey) [inline app/src/main/java/com/mealsmadeeasy/ui/home/discover/DiscoverFragment.kt]: Why is this no longer private?

File: app/src/main/java/com/mealsmadeeasy/ui/home/discover/FilterBottomDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import com.mealsmadeeasy.R
+import com.mealsmadeeasy.data.MealStore
+import com.mealsmadeeasy.model.Filter
+import javax.inject.Inject
+
+private const val KEY_SAVED_SEL_FILTERS = "FilterFrag.SELECTED_FILTERS"
+
+class FilterBottomDialogFragment : BottomSheetDialogFragment() {
+    @Inject
+    lateinit var mealStore: MealStore
```
Comment (andrewbailey) [inline app/src/main/java/com/mealsmadeeasy/ui/home/discover/FilterBottomDialogFragment.kt]: Nit: Let's keep `@Inject` on the same line as the variable.

File: app/src/main/res/menu/activity_search.xml [hunk 3]
```diff
@@ -9,5 +9,9 @@
         android:title="@string/search_bar_title"
         app:showAsAction="always" />
 
-
+    <item
+        android:id="@+id/menu_recipe_filter"
+        android:icon="@drawable/ic_filter_list_white_24dp"
+        android:title="@string/filter_icon_title"
+        app:showAsAction="ifRoom" />
```
Comment (andrewbailey) [inline app/src/main/res/menu/activity_search.xml]: Let's make this a FAB instead.

File: app/src/main/res/layout/view_filter_category.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content" >
+
+    <CheckedTextView
+        android:id="@+id/filter_name"
+        android:padding="@dimen/filter_category_textview_padding"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:checkMark="?android:attr/listChoiceIndicatorMultiple" />
```
Comment (andrewbailey) [inline app/src/main/res/layout/view_filter_category.xml]: w h a t. this is a thing? Can you make the checkmark appear on the left?

File: app/src/main/res/layout/fragment_filter.xml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    android:layout_height="wrap_content"
+    android:orientation="vertical">
+
+    <TextView
+        android:id="@+id/filter_bottom_sheet_heading"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/filter_frag_heading_height"
+        android:gravity="center"
+        android:text="@string/filter_bottom_sheet_title"
+        android:textSize="@dimen/filter_frag_heading_size" />
```
Comment (andrewbailey) [inline app/src/main/res/layout/fragment_filter.xml]: I don't think this needs this since there are no other options besides filters. We can keep it, but consider making it left aligned and making the style of it `TextAppearance.OtherStuffIForgetThatStudioWillAutocomplete.Subhead`.

**Review by andrewbailey — CHANGES_REQUESTED** [review]
Okay. So it looks okay, except that I can tell you didn't read [the documentation on filters](https://github.com/MealsMadeEasy/Backend/wiki/Search#mealssearchfilters).

TL;DR Edamam filters are mutually exclusive from each other. So the UI needs to represent that. May I suggest a RecyclerView for FilterGroups?

**Review by andrewbailey — APPROVED** [review]
